test/integration/integration_test.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+require_relative 'integration_helper'
+
+IntegrationTest.describe 'Integration testing with Rails app', '>= 2.5.3' do
+  include ShellHelper
+
+  RAILS_APP_PATH = File.expand_path('fakeapp', __dir__).freeze
+
+  before(:all) do
+    Dir.chdir(RAILS_APP_PATH) { sh('bundle install') }
+  end
+
+  around do |test|
+    Dir.chdir(RAILS_APP_PATH) { test.call }
+  end
+
+  it 'prints simple progress' do
+    sh("#{Bundler.root}/bin/bumbler").must_include <<~TEXT
+      ( 0/10)  rails
+      ( 1/10)  spring
+      ( 1/10)  spring-watcher-listen
+      ( 1/10)  listen
+      1 of 10 gems required
```

Comment:
there are two seperate weird things going on. when bootsnap is running, we don't get the initial gemfile requires. it doesn't totally wipe out the hook, but something else gets screwed up. also these remaining gem loads don't increment the count properly.